lib/utils/schema.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const _ = require('lodash')
+const Joi = require('joi')
+const Hoek = require('hoek')
+
+module.exports = function() {
+    const toJoiPostValidatation = function(schema) {
```

Comment:
As far as I understand that change was made to allow a single resource to return instances of multiple types. 

In our use case here at AGCO you could have GET /users return dealerUsers, agcoUsers, ownerUsers etc. This facilitates linking.